waiter/src/waiter/util/http_utils.clj

```diff
@@ -38,9 +38,8 @@
 
 (def ^Base64 base64 (Base64.))
 
-(defn spnego-authentication
-  "Returns an Authentication$Result for endpoint which will use SPNEGO to generate an Authorization header"
-  [^URI endpoint]
+(defn create-spnego-authentication-result-factory
+  [^URI endpoint proxy-auth?]
```

Comment:
I think we should pass the authorization header name instead of `proxy-auth?` boolean value from the caller and avoid the if-else branching in the code below.